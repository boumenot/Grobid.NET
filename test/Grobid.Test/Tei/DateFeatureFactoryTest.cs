using System;
using System.Linq;
using System.Xml.Linq;

using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalUtilities.Utilities;
using Xunit;

using Grobid.NET;
using Grobid.NET.Feature;
using Grobid.NET.Feature.Date;
using Grobid.Test.Feature;

namespace Grobid.Test.Tei
{
    [UseReporter(typeof(DiffReporter))]
    public class DateFeatureFactoryTest
    {
        [Fact]
        public void DateFeatureFactoryTest00()
        {
            var tei = @"<dates><date><month>March</month> <day>28</day>, <year>2007</year></date></dates>";

            var factory = new DateFeatureVectorFactory(
                new FeatureExtractor(EmptyLexicon.Instance));

            var formatter = new LabeledFeatureFormatter<DateFeatureVector>(
                new DateFeatureFormatter());

            var testSubject = new DateFeatureFactory(factory);
            Approvals.VerifyAll(
                testSubject.Create(XDocument.Parse(tei)),
                x => String.Join("\n", x.Select(y => formatter.Format(y.FeatureVector, y).Join(" "))) + "\n");
        }

        [Fact]
        public void DateFeatureFactoryTest01()
        {
            var tei = @"<dates>
  <date><month>March</month> <day>28</day>, <year>2007</year></date>
  <date>Published <day>25</day> <month>May</month> <year>2011</year></date>
</dates>";

            var factory = new DateFeatureVectorFactory(
                new FeatureExtractor(EmptyLexicon.Instance));

            var formatter = new LabeledFeatureFormatter<DateFeatureVector>(
                new DateFeatureFormatter());

            var testSubject = new DateFeatureFactory(factory);
            Approvals.VerifyAll(
                testSubject.Create(XDocument.Parse(tei)),
                x => String.Join("\n", x.Select(y => formatter.Format(y.FeatureVector, y).Join(" "))) + "\n");
        }
    }
}
