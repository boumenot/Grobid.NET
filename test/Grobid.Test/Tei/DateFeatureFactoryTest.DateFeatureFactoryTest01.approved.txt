March march M Ma Mar Marc h ch rch arch LINESTART INITCAP NODIGIT 0 0 1 NOPUNCT I-<month>
28 28 2 28 28 28 8 28 28 28 LINEIN NOCAPS ALLDIGIT 0 0 0 NOPUNCT I-<day>
, , , , , , , , , , LINEIN ALLCAP NODIGIT 1 0 0 COMMA I-<other>
2007 2007 2 20 200 2007 7 07 007 2007 LINEEND NOCAPS ALLDIGIT 0 1 0 NOPUNCT I-<year>

Published published P Pu Pub Publ d ed hed shed LINESTART INITCAP NODIGIT 0 0 0 NOPUNCT I-<other>
25 25 2 25 25 25 5 25 25 25 LINEIN NOCAPS ALLDIGIT 0 0 0 NOPUNCT I-<day>
May may M Ma May May y ay May May LINEIN INITCAP NODIGIT 0 0 1 NOPUNCT I-<month>
2011 2011 2 20 201 2011 1 11 011 2011 LINEEND NOCAPS ALLDIGIT 0 1 0 NOPUNCT I-<year>

